### ISSUE TITLE
SSL Error (Swift)

### ISSUE THREAD

**Issue by MRodSebastian** [socketio/socket.io-client-swift#705]
Hi, I'm working with this library and i have a problem when I create a new socket. 
```Swift
 let socket = SocketIOClient(socketURL: URL(string: socketURL)!, config: [
            .log(true),
            .connectParams(["access_token": access_token ?? ""]),
            .forcePolling(true),
            .forceNew(true),
            .selfSigned(true),
            .secure(true),
            .sessionDelegate(sessionManager)
        ])
```

### Console log:

NSURLSession/NSURLConnection HTTP load failed (kCFStreamErrorDomainSSL, -9806)
ERROR SocketEnginePolling: An SSL error has occurred and a secure connection to the server cannot be made.
ERROR SocketEngine: An SSL error has occurred and a secure connection to the server cannot be made.
ERROR SocketIOClient: An SSL error has occurred and a secure connection to the server cannot be made.
LOG SocketIOClient: Handling event: error with data: ["An SSL error has occurred and a secure connection to the server cannot be made."]
LOG SocketIOClient: Handling event: statusChange with data: [SocketIO.SocketIOClientStatus]

**Comment by nuclearace** [comment]
Are you handling the self signed cert in your session delegate?

**Comment by ManuRodriguezSebastian** [comment]
Yes, im using this method in mi sessionManager

````swift
func urlSession(_ session: URLSession, didReceive challenge: URLAuthenticationChallenge, completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {
        if challenge.protectionSpace.authenticationMethod == NSURLAuthenticationMethodServerTrust {
            let serverTrust = challenge.protectionSpace.serverTrust
            completionHandler(Foundation.URLSession.AuthChallengeDisposition.useCredential, URLCredential(trust: serverTrust!))
        }
    }
````

But it's never called...

**Comment by EmotionV** [comment]
i have same question

**Comment by GDXRepo** [comment]
Same issue. Any solution by far? Please give the valid algorithm to connect to secured server with self-signed certificate.

**Comment by erversteeg** [comment]
@MRodSebastian Ensure that sessionManager is a URLSessionDelegate 
`class SessionManager: NSObject, URLSessionDelegate`


With this constructor,

`manager = SocketManager(socketURL: URL(string: "https://192.168.0.100:5010")!, config: [.log(true), .compress, .selfSigned(true), .sessionDelegate(self)]) ` (In my case I used the same class as delegate).

and delegate method

```
func urlSession(_ session: URLSession, didReceive challenge: URLAuthenticationChallenge, completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {
        completionHandler(.useCredential, URLCredential(trust: challenge.protectionSpace.serverTrust!))
    }
```

I was able to get connected to a local server using a self-signed certificate after receiving a code=1202 error.  However I kept getting an .on(.error) response and saw in the server logs that the swift client was unsupported and after updating Socket.io-Client-Swift to 16.0.0 (from 15.0.2) everything started working, .on(.connect) and other .on events.
